scripts/run-percy.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-console */
+
+const multimatch = require('multimatch');
+const shelljs = require('shelljs');
+
+const getChangedFiles = range => {
```

Comment:
In case it helps, this is how we used to do it on CircleCI:

```bash
hasChangesInMatchingFiles() {
  # Extract commit range
  # (`cut -d/ -f7` means "split by `/` and pick the 7th field in the list")
  # E.g. https://github.com/commercetools/merchant-center-frontend/compare/1945a7a7a011...3ec9fece99cf
  #   => 1945a7a7a011...3ec9fece99cf
  COMMIT_RANGE=$(echo "${CIRCLE_COMPARE_URL}" | cut -d/ -f7)

  # NOTE: unfortunately in case of force pushes (e.g. rebases) the commit range is useless
  # as the checked out code does not contain the original commit.
  # The error you would see contains `unknown revision or path not in the working tree.`
  # https://github.com/travis-ci/travis-ci/issues/2668
  # In this case we check if git considers the value to be a valid range. If that's the
  # case the evaluated value would be `0`, otherwise something like `128`.
  IS_COMMIT_RANGE_VALID=$(git diff "$COMMIT_RANGE" &> /dev/null; echo $?)

  # Furthermore, we check if the range is actually a range, as sometimes the `CIRCLE_COMPARE_URL`
  # can be a single commit. In this case we also consider it to not be a valid range.
  if [[ $COMMIT_RANGE != *"..."* ]]; then
    IS_COMMIT_RANGE_VALID=9 # some custom exit code that is not `0`
  fi

  if [[ $IS_COMMIT_RANGE_VALID != 0 ]]; then
    echo "CircleCI commit range diff failed, probably due to a new PR or a force push (rebase), falling back to single commit"
    if git diff-tree --no-commit-id --name-only -r "$CIRCLE_SHA1" |grep --quiet -E "$MATCHING_PATH_TRIGGERS"; then
      return 0
    else
      return 1
    fi
  else
    if git diff --no-commit-id --name-only "$COMMIT_RANGE" |grep --quiet -E "$MATCHING_PATH_TRIGGERS"; then
      return 0
    else
      return 1
    fi
  fi
}
```

Reply:
Didn't we have a lot of problems with that though and ended up removing it?

Reply:
We removed it for other reasons. My point was more about the edge cases to consider when trying to extract the commit logs.

Reply:
Yeah I think I'll need some help with this part